### PR TITLE
Fix deserialization of events with entitlements and add better error logging

### PR DESCRIPTION
**Pull request by lealobanov**
Resolves issue #66 

## Release Notes

- **New Features**
	- Introduced a new example for debugging transaction fetching by ID.
	- Added a new JSON structure for event capabilities, enhancing metadata representation.
	- Added a new test for account capability event encoding and decoding.

- **Bug Fixes**
	- Improved error handling and logging for event deserialization and transaction results.

- **Chores**
	- Updated `.gitignore` to include `myenv` for better environment management.
	- Renamed `AuthAccountKind` to `AccountKind` for clarity in naming conventions.

### CODE REVIEW

File: myenv/bin/activate.csh
Comment (lealobanov) [reply myenv/bin/activate.csh:11]: Updated gitignore, will delete `myenv` before merging

File: examples/transactions_examples.py
Comment (lealobanov) [inline examples/transactions_examples.py:286]: Added example which makes a mainnet call to the failing tx ID `ef6e6dcba87c5f853e0ec940dd382d609ffe11109c28ed239843d71618fa1d71`, will remove this before merging changes

File: failed_events.json
Comment (lealobanov) [inline failed_events.json:1]: This is an output of the failing event payloads for reference, will remove before merging changes

File: flow_py_sdk/client/entities.py
Comment (lealobanov) [reply flow_py_sdk/client/entities.py]: Updated

File: flow_py_sdk/client/entities.py
Comment (janezpodhostnik) [reply flow_py_sdk/client/entities.py:2]: This Is a good idea ^

**Review by janezpodhostnik — COMMENTED** [review]
Thank you very much for the contribution!

Can you add a short explanation of why the events failed to de-serialize, I'm not quite sure I get it. Was there an unexpected dict/list in the event? 

As for the error handling: 
- I very much like the Idea of extra logging, that will make it a lot easier to figure out whats wrong next time.
- I'm not sure I like the idea of just logging and continuing. That seems very dangerous. I would rather the error was logged and the exception re-thrown. There could be an optional parameter where instead of re-throwing it would just return the json object if decoding failed. What do you think?

File: flow_py_sdk/client/entities.py
Comment (lealobanov) [reply flow_py_sdk/client/entities.py:326]: @janezpodhostnik initially when debugging I considered introducing a max limit on number events processed per transaction as I thought it might be causing the issue... in the end it didn't seem to have much impact on this specific transaction so I opted not to pursue these changes for now.

Comment (lealobanov) [reply flow_py_sdk/client/entities.py:326]: Will resolve this comment for time being and we can look into introducing a restriction on event count if we run into further issues

**Comment by lealobanov** [comment]
Hi @janezpodhostnik ,

- Just updated the exception handling to raise an exception and exit the decoding if one of the events failed to decode (instead of continuing).

I think the source of the bug might have been the recent migration to Cadence 1.0, as I noticed all of the events which failed to decode had complex nested dict/list fields such as `authorizations` and `staticType` found in the `EntitlementConjunctionSet` class.  Entitlements seem to be a new syntax introduced since the C1.0 release (https://www.youtube.com/watch?v=aZ1aCBtPpXo), so I suspect this might be why this wasn't throwing an error before.

Thanks!

File: flow_py_sdk/cadence/decode.py
Comment (franklywatson) [reply flow_py_sdk/cadence/decode.py:57]: @lealobanov what do we think about this suggestion? Given decode is a heavily used method I would err on making ti as robust as we can

File: flow_py_sdk/client/entities.py
Comment (janezpodhostnik) [inline flow_py_sdk/client/entities.py:163]: I like the addition of logging, but I would not return None, as it can lead to confusing results for users (everything seems ok, but no value is returned). I would instead just re-raise the exception here. n

same goes for all try-catch blocks in this file: `flow_py_sdk/client/entities.py`.

**Comment by janezpodhostnik** [comment]
> Just updated the exception handling to raise an exception and exit the decoding if one of the events failed to decode (instead of continuing).

Looks great, thank you. Can you do the same for `flow_py_sdk/client/entities.py`

I did some [digging](https://github.com/janezpodhostnik/flow-py-sdk/pull/68). And I realized that the recursive decoding should not be needed, since the cadence [json api objects](https://cadence-lang.org/docs/json-cadence-spec) already define if something can be a list or not, and the `EntitlementsKind` do correctly accept a list.

```py
class EntitlementsKind(Kind, ABC):
    def __init__(self, entitlements: list[EntitlementBaseKind]) -> None:
```

The error was that an element that should have been decoded was not due to the incorrect naming of "Account" (it was still "AuthAccount" since I forgot to change it from before cadence 1.0)

Changing that fixed the issue without the need of adding the recursive decoding.


If you want to pull those changes in your PR and cleanup the temporary things we can get this merged.

As for this:

> Some other preventative measures we could take on the decoding of events array:
> 
>     Imposing a max number of events that get decoded per transaction
>     Imposing a max limit on event payload size

That is a great idea, and the contribution would be much appreciated, but lets not add it to this PR to keep it smaller and because its supposed to be a bugfix.

**Comment by lealobanov** [comment]
Hi @janezpodhostnik ,

Good catch, I didn't notice that the naming of the field had changed - thank you so much for looking into it. I've just pulled in your branch and cleaned up the test code. I've run the latest changes against tx ID `ef6e6dcba87c5f853e0ec940dd382d609ffe11109c28ed239843d71618fa1d71` and can confirm it decoded successfully, so have gone ahead and removed this from the transaction_examples files.

File: flow_py_sdk/cadence/decode.py
Comment (lealobanov) [reply flow_py_sdk/cadence/decode.py]: Incorporated these improved messages but left the status code returned as error

File: flow_py_sdk/cadence/__init__.py
Comment (lealobanov) [reply flow_py_sdk/cadence/__init__.py:93]: Can confirm simple_kinds.py has already been updated with these changes

File: flow_py_sdk/client/entities.py
Comment (lealobanov) [reply flow_py_sdk/client/entities.py:168]: Will hold off on these improvements in a separate PR

**Review by janezpodhostnik — APPROVED** [review]
Looks great! Thank you for the contribution.

**Pull request merged**
